File: frontend/src/types/dto/Compra.ts
export interface DetalleCompra {
  idProducto: number
  producto: string
  cantidad: number
  costoUnitario: number
}

// Corresponde al objeto 'content' de la respuesta paginada
export interface Compra {
  idCompra: number
  total: number
  fechaHora: string
  proveedor: string
  usuario: string
  estadoCompra: string
  detalles: DetalleCompra[]
}

export interface PaginaDeCompras {
  content: Compra[]
  totalPages: number
  totalElements: number
  number: number 
  size: number
}

export interface CompraDTO {
  idProveedor: number
  detalles: {
    idProducto: number
    cantidad: number
    costoUnitario: number
  }[]
}

File: backend/src/main/java/com/example/negocio/entity/Compra.java
package com.example.negocio.entity;

import com.example.negocio.enums.EstadoCompra;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Data @Builder @AllArgsConstructor @NoArgsConstructor
@Entity @Table
public class Compra {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idCompra;
    private BigDecimal total;
    private LocalDateTime fechaHora;

    @Enumerated(EnumType.STRING)
    private EstadoCompra estadoCompra;

    @ManyToOne @JoinColumn(name = "idProveedor")
    private Proveedor proveedor;

    @OneToMany(mappedBy = "compra", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference
    private List<DetalleCompra> detalles;

    @ManyToOne @JoinColumn(name = "idUsuario")
    private Usuario usuario;
}
